src/components/DiagramFrame/DiagramFrame.tsx

```diff
@@ -0,0 +1,267 @@
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import React, { memo } from "react"
+import {
+  SpaceVertical,
+  IconButton,
+  Layout,
+  SelectOptionProps,
+} from "@looker/components"
+import { SelectionInfoPacket, VisibleViewLookup } from "../interfaces"
+import { useCurrentModel } from "../../utils/routes"
+import { DiagrammedModel, DiagramMetadata } from "../../utils/diagrammer"
+import "./styles.css"
+import {MetadataPanel} from "./MetadataPanel/MetadataPanel"
+import { DiagramSettings } from "./DiagramSettings"
+import { HelpPanel } from "./HelpPanel"
+import { ViewOptions } from "./ViewOptions"
+import { DiagramHeader } from "./DiagramHeader"
+import { DiagramCanvas } from "./DiagramCanvas/DiagramCanvas"
+import { 
+  ZOOM_INIT,
+  X_INIT,
+  Y_INIT,
+  OVERRIDE_KEY,
+  OVERRIDE_KEY_SUBTLE
+} from '../../utils/constants'
+import { ILookmlModel, ILookmlModelExplore, IGitBranch } from "@looker/sdk/lib/sdk/4.0/models"
+import {DiagramFrameProps, ExploreDropdown, Rail, Stage} from "./types"
+import {getBranchOptions} from "./utils"
+
+export const DiagramFrame: React.FC<DiagramFrameProps> = ({
+  unfilteredModels,
+  pathModelName,
+  pathExploreName,
+  modelDetail,
+  dimensions,
+  modelError,
+  setModelError,
+  hiddenToggle,
+  setHiddenToggle,
+  displayFieldType,
+  setDisplayFieldType,
+  selectedBranch,
+  setSelectedBranch,
+  }) => {
+  const [viewVisible, setViewVisible] = React.useState<VisibleViewLookup>({})
+  const [showSettings, setShowSettings] = React.useState(true)
+  const [showHelp, setShowHelp] = React.useState(false)
+  const [showViewOptions, setShowViewOptions] = React.useState(false)
+  const [showNoAside, setShowNoAside] = React.useState(false)
+  const [reload, setReload] = React.useState(false)
+  const [selectionInfo, setSelectionInfo] = React.useState<SelectionInfoPacket>({})
+  const [zoomFactor, setZoomFactor] = React.useState(ZOOM_INIT)
+  const [viewPosition, setViewPosition] = React.useState({x: X_INIT, y: Y_INIT})
+  const [minimapEnabled, setMinimapEnabled] = React.useState(false)
+  const [minimapUntoggled, setMinimapUntoggled] = React.useState(true)
+  const currentModel = useCurrentModel(selectedBranch, modelError)
+
+  const handleHiddenToggle = (event: any) => setHiddenToggle(event.target.checked)
+
+  function showDiagram() {
+    setShowHelp(false)
+    setShowViewOptions(false)
+    setShowSettings(false)
+    setShowNoAside(true)
+    return true
+  }
+
+  function toggleExploreInfo() {
+    if (currentExplore && modelDetail) {
+      Object.keys(selectionInfo).length === 0 || selectionInfo.lookmlElement !== "explore"
+      ? setSelectionInfo({
+        lookmlElement: "explore"
+      })
+      : setSelectionInfo({})
+    }
```

Comment:
Is this equivalent?  I'm not sure what other keys on `selectInfo` matter here, but I found the current implementation hard to read.

```suggestion
      setSelectionInfo({
        lookmlElement: selectionInfo.lookmlElement !== "explore" ? "explore" : undefined
      })
    }
```